tests/spread/extensions/env-injector/task.yaml

```diff
@@ -0,0 +1,53 @@
+summary: Build and run a basic hello-world snap using extensions
+
+systems:
+  - ubuntu-24.04
+
+environment:
+
+  SNAPCRAFT_ENABLE_EXPERIMENTAL_EXTENSIONS: "1"
+  SNAP_DIR: ../snaps/env-injector-hello
+  SNAP: env-injector-hello
+
+prepare: |
+
+  #shellcheck source=tests/spread/tools/snapcraft-yaml.sh
+  . "$TOOLS_DIR/snapcraft-yaml.sh"
+  set_base "$SNAP_DIR/snap/snapcraft.yaml"
+
+restore: |
+
+  cd "$SNAP_DIR"
+  snapcraft clean
+  rm -f ./*.snap
+  rm -rf ./squashfs-root
+  #shellcheck source=tests/spread/tools/snapcraft-yaml.sh
+  . "$TOOLS_DIR/snapcraft-yaml.sh"
+  restore_yaml "snap/snapcraft.yaml"
+
+execute: |
+
+  cd "$SNAP_DIR"
+  snapcraft
+
+  unsquashfs "${SNAP}"_1.0_*.snap
+
+  # Check that the env-exporter program is present
+  [ -f squashfs-root/bin/command-chain/env-exporter ]
+  
+  # Check that the hello command is present
+  [ -f squashfs-root/usr/local/bin/hello ]
+
+  snap install "${SNAP}"_1.0_*.snap --dangerous
+
+  # Create envfile
+  echo 'HELLO_WORLD="Hello, World"' >> envfile.env
+
+  # Set env vars: Global, App specific, and envfile
+  snap set env-injector-hello env.hello="Hello"
+  snap set env-injector-hello apps.hello.env.world="World"
+  snap set env-injector-hello envfile="${SNAP_DIR}"/enfile.env
```

Comment:
Other needed test cases:
- snap options keys with hyphen converted to equivalent
- `<env>` key with dot rejected (ignored with error message): e.g. `env.hello.world`
- Order of exporting variables (those from env files overridden by snap options)

Reply:
Addressed! Thanks.